code/modules/mob/mob_helpers.dm

```diff
@@ -326,22 +326,17 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
 #define PIXELS_PER_STRENGTH_VAL 16
 
 /proc/shake_camera(mob/M, duration, strength=1)
-	set waitfor = 0
-	if(!M || !M.client || M.shakecamera || M.stat || isEye(M) || isAI(M))
+	var/current_time = world.time
+	if(!M || !M.client || (M.shakecamera > current_time)|| M.stat || isEye(M) || isAI(M))
 		return
-	M.shakecamera = TRUE
+	M.shakecamera = current_time + min(TICKS_PER_RECOIL_ANIM, duration)
```

Comment:
This should be `(steps + 1) * TICKS_PER_RECOIL_ANIM` I think, otherwise you're only delaying until the end of the first shake.

Reply:
Negative. The amount of steps are based off the duration so this is the correct way to it
`var/steps = min(1, Floor(duration/TICKS_PER_RECOIL_ANIM))-1`

Reply:
there we go should be fixed. the steps are based off the duration so it should be correct